tests/test_generate.py

```diff
@@ -214,5 +214,34 @@ def test_set_plain_language_summary(self):
         self.assertEqual(ElementTree.tostring(parent_tag), expected)
 
 
+class TestSetDatasets(unittest.TestCase):
+
+    def test_set_datasets(self):
+        parent_tag = Element('root')
+        expected = (
+            b'<root>'
+            b'<Object Type="Dryad"><Param Name="id">10.5061/dryad.example</Param></Object>'
+            b'<Object Type="NCBI:geo"><Param Name="id">GSE48760</Param></Object>'
+            b'</root>')
+        article = Article()
+        dataset = Dataset()
+        dataset.uri = '10.5061/dryad.example'
+        dataset.doi = '10.5061/dryad.example'
+        article.datasets.append(dataset)
+        # first ref is a duplicate should be ignored
```

Comment:
is this is what is being tested? if so, stick it in the method's docstring.

Reply:
I set out to test a bunch of values and then settled on testing both dataset and ref data with one duplicate. It is one test case to capture any edge cases that do not exist in the `00666` article XML. I'll describe it better, and I'm going to expand it to demonstrate the (soon to be) case-insensitive matching of ref source values.